src/snek5000/make.py

```diff
@@ -128,3 +133,98 @@ def exec(self, *rules, dryrun=False, keep_incomplete=True, **kwargs):
                 log_handler=self.log_handler,
                 **kwargs,
             )
+
+
+class _Nek5000Make(Make):
+    """Snakemake interface to build Nek5000 tools and other dependencies.
+    This class would prevent unnecessary rebuild of Nek5000 if there is no
+    change in the compiler configuration.
+
+    .. note::
+
+        There is a small caveat with this solution. If a series of jobs are
```

Comment:
This note is interesting. Maybe a warning note could be added at the end of https://snek5000.readthedocs.io/en/latest/configuring.html?